code/modules/mob/living/bot/cleanbot.dm

```diff
@@ -1,8 +1,20 @@
+// Updated by Nadrew, bits and pieces taken from Baycode, but fairly heavily modified to function here (and because a few bits of the baycode was ehh)
+
+// The main issue in the old code was the Life() loop and the fact that it could go infinite really easily.
+// The fix involved labeling the various loops involved so they could be continued and broken properly.
+// It also decreases the amount of calls to AStar() and handle_target()
+
+var/list/cleanbot_types // Going to use this to generate a list of types once then cull it out locally, see comments below for more info
+
+/obj/effect/decal/cleanable/var
+	being_cleaned = 0
+	tmp/mob/living/bot/cleanbot/clean_marked = 0 // If a cleaning bot has marked the cleanable to be cleaned, to prevent multiples from going to the same one.
```

Comment:
This should not interrupt the BYOND garbage collection process, aye? Or should this be forcibly set to `null` during Destroy/before calling qdel() on it?